### PR TITLE
Snapshot clean-ups

### PR DESCRIPTION
**Pull request by nnethercote**
Two minor clean-ups involving snapshots.

### CODE REVIEW

File: src/librustc/infer/region_constraints/leak_check.rs [hunk 1]
```diff
@@ -13,19 +13,11 @@ impl<'tcx> RegionConstraintCollector<'tcx> {
     /// This is a temporary backwards compatibility measure to try and
     /// retain the older (arguably incorrect) behavior of the
     /// compiler.
-    ///
-    /// NB. The use of snapshot here is mostly an efficiency thing --
-    /// we could search *all* region constraints, but that'd be a
-    /// bigger set and the data structures are not setup for that. If
-    /// we wind up keeping some form of this check long term, it would
-    /// probably be better to remove the snapshot parameter and to
-    /// refactor the constraint set.
```
Comment (varkor) [inline src/librustc/infer/region_constraints/leak_check.rs]: Was the constraint set ever refactored? This comment seems odd in light of the snapshot parameter already being unused.

Comment (nnethercote) [reply src/librustc/infer/region_constraints/leak_check.rs]: I don't know. I'm not at all familiar with this function. I figured the comment was obviously wrong and removing it seemed the best choice.

Comment (varkor) [reply src/librustc/infer/region_constraints/leak_check.rs]: cc @nikomatsakis

Comment (nnethercote) [reply src/librustc/infer/region_constraints/leak_check.rs]: I see. I have rewritten to the first patch; now it just clarifies that comment. The second patch still makes sense.

**Comment by nnethercote** [comment]
New code is up.

r? @nikomatsakis

**Review by nikomatsakis — APPROVED** [review]
Thanks @nnethercote

**Pull request merged**